### PR TITLE
API Documentation

### PR DESCRIPTION
**Pull request by njseeto**
Story: https://trello.com/c/VsDaNKQi/1567-backend-create-editor-api-endpoints-for-default-text

### Add API documentation via Redocly
Redocly has been chosen as is is currently being used for the Form Builder API docs.

### Add dockerignore file
We don't need to add documentation into the app containers

### Add github action to publish github pages

### CODE REVIEW

File: docs/docs/index.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>API Reference | ReDoc</title>
```
Comment (brenetic) [inline docs/docs/index.html:4]:
```suggestion
    <title>MoJ Forms Editor API Docs</title>
```

File: docs/openapi/components/responses/403.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    description: Response message
+    type: array
+    items:
+      type: string
+      minLength: 1
+      example: You do not have permission to perform this operation
+example: {
+  message: [
+    'Unauthorised'
+  ]
```
Comment (brenetic) [inline docs/openapi/components/responses/403.yaml:14]: is this an array of strings?

File: docs/openapi/paths/api@services@{service_id}@pages@{pages_uuid}.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    schema:
+      type: string
+  - name: page_uuid
+    in: path
+    description: The Page UUID
+    required: true
+    schema:
+      type: string
+  security:
+    - jwt: []
```
Comment (brenetic) [inline docs/openapi/paths/api@services@{service_id}@pages@{pages_uuid}.yaml:21]: at this point we won't be using jwt as the security mechanism. i am unsure how to represent the session type of security in openapi . . .

File: docs/openapi/components/responses/403.yaml (hunk 2 above)
Comment (njseeto) [reply docs/openapi/components/responses/403.yaml:14]: ah yes, good spot.

File: docs/openapi/paths/api@services@{service_id}@pages@{pages_uuid}.yaml (hunk 3 above)
Comment (njseeto) [reply docs/openapi/paths/api@services@{service_id}@pages@{pages_uuid}.yaml:21]: perhaps i can change to `basicAuth: []` ?

**Comment by brenetic** [comment]
good stuff @njseeto . can you also add the /docs folder to dependabot so it can notify us of dependency updates for redocly

File: .dependabot/config.yml [hunk 4]
```diff
@@ -18,6 +18,15 @@ update_configs:
     - tomas-stefano
     - brenetic
     - njseeto
+  - package_manager: "javascript"
+    directory: "/docs"
+    update_schedule: "live"
+    default_reviewers:
+    - mattempty
+    - form-builder-developers
```
Comment (brenetic) [inline .dependabot/config.yml:26]:
```suggestion
```

**Pull request closed without merging**